ipamutils/utils.go

```diff
@@ -48,3 +64,62 @@ func initGranularPredefinedNetworks() []*net.IPNet {
 	}
 	return pl
 }
+
+// InitAddressPools allows to initialize the local and global scope predefined
+// address pools to the desired values. It fails is invalid input is passed
+// or if the predefined pools were already initialized.
+func InitAddressPools(list []*PredefinedPools) error {
+	localPools := make([]*net.IPNet, 0, len(list))
+	//done := false
+
+	for _, p := range list {
+		if p == nil {
+			continue
+		}
+		_, b, err := net.ParseCIDR(p.Base)
+		if err != nil {
+			return fmt.Errorf("invalid base pool %q: %v", p.Base, err)
+		}
+		ones, _ := b.Mask.Size()
+		if p.Size <= 0 || p.Size < ones {
+			return fmt.Errorf("invalid pools size: %d", p.Size)
+		}
+		localPools = append(localPools, initPools(p.Size, b)...)
+	}
+	if localPools == nil {
+		localPools = initBroadPredefinedNetworks()
+	}
+	PredefinedBroadNetworks = localPools
+	return nil
+}
+
+func initPools(size int, base *net.IPNet) []*net.IPNet {
+	one, bits := base.Mask.Size()
+	mask := net.CIDRMask(size, bits)
+	n := 1 << uint(size-one)
+	s := uint(bits - size)
+	list := make([]*net.IPNet, 0, n)
+
+	for i := 0; i < n; i++ {
+		ip := copyIP(base.IP)
+		addIntToIP(ip, uint(i<<s))
+		list = append(list, &net.IPNet{IP: ip, Mask: mask})
+	}
+
+	return list
+}
+
+func copyIP(from net.IP) net.IP {
+	ip := make([]byte, len(from))
+	for i := 0; i < len(ip); i++ {
+		ip[i] = from[i]
+	}
+	return ip
+}
+
+func addIntToIP(array []byte, ordinal uint) {
```

Comment:
Do we support IPv6 in these configurations?  I ask because net.IP is supposed to represent either an IPv4 or an IPv6 address.  The copyIP() routine is generic enough to handle both, but addIntToIP isn't.  If the network mask starts above 32 bits in the address, then this simply fails.  We'd need to modify the parameter to either be a []byte to "add/or" or give it both an "add" and "shift" parameter.  This assumes we are bounding the number of networks per comment above.

Reply:
We dont support IPV6 right now.